kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NetworkPropertiesTest.java

```diff
@@ -0,0 +1,192 @@
+package org.eclipse.kura.nm.configuration;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.junit.Test;
+
+public class NetworkPropertiesTest {
+
+	NetworkProperties netProps;
+	Map<String, Object> properties;
+
+	@Test(expected = NullPointerException.class)
+	public void shouldFailWhenNull() {
+		givenTheFollowingProperties(null);
+		whenNetworkPropsIsCreated();
+	}
+
+	@Test
+	public void shouldReturnPropertiesWithGetProperties() {
+		givenTheTestProperties();
+		whenNetworkPropsIsCreated();
+		thenCompareGetPropertiesToPassedProperties();
```

Comment:
I like where this is going but take a look here for a more flexible approach: https://github.com/eclipse/kura/blob/develop/kura/test/org.eclipse.kura.ai.triton.server.test/src/test/java/org/eclipse/kura/ai/triton/server/TritonServerServiceOptionsTest.java

With `givenTheTestProperties` the reader of the code must go to the bottom of the file to understand what's the content of the Map. If you go for the approach of the link I just gave you you would have something like:

```suggestion
		givenMapWith("testKey1", "testString1");
		givenMapWith("testKeyNull", null);
		givenNetworkPropertiesBuiltWith(this.map);
		
		then...
```

which is more readable and flexible